msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AcquireTokenByInteractiveFlowSupplier.java

```diff
@@ -209,7 +209,7 @@ private AuthorizationResult getAuthorizationResultFromHttpListener() {
                 expirationTime = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()) + 1;
             }
 
-            while (result == null && !interactiveRequest.futureReference().get().isCancelled()) {
+            while (result == null && !interactiveRequest.futureReference().get().isDone()) {
```

Comment:
Could you add a comment on what "isDone" is doing here? I think it returns true when cancel() is called, or when the future completes with success?

Reply:
[isDone](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#isDone--) returns true when a future is cancelled (like the current behavior), plus when it completes normally or with an exception. In other words, `isDone` covers any situation where we can't return the auth result, and there's no point in continuing this loop.

By only checking `isCancelled` there were scenarios where thread spun off to complete futures could continue to pointlessly run (such as when the app developers sets a timeout in the future https://github.com/AzureAD/microsoft-authentication-library-for-java/issues/741).